vignettes/function-flow.Rmd

```diff
@@ -0,0 +1,161 @@
+
+<!-- TODO: Which specific ICD 8 and 10 codes are included? -->
+
+### Diabetes-specific podiatrist services
+
+The function `include_podiatrist_services()` uses `sysi` or `sssy` as
+input to extract the dates of all diabetes-specific podiatrist services.
+
+<!-- TODO: Add details on how this filtering should be done -->
+
+### GLD purchases
+
+The function `include_gld_purchases()` uses `lmdb` to extract the dates
+of all GLD purchases (from 1997 onwards).
+
+<!-- TODO: Add details on how this filtering should be done -->
+
+<!-- TODO: Add this + link to resource "For details about this, see [link]." -->
+
+## Exclusion events
+
+### HbA1c tests and GLD purchases during pregnancy
+
+The function `exclude_pregnancy()` uses diagnoses from LPR2 or LPR3 as
+input and is used to exclude both HbA1c tests and GLD purchases during
+pregnancy.
+
+Internally, this relies on the function `get_pregnancy_dates()` that
+contains the following three helper functions:
+
+-   `calculate_pregnancy_index_date_for_mc_visits_wo_end_date()` (this
+    might be removed with the inclusion of the birth register)
+-   `get_pregnancy_end_dates()`: Keep maternal care visits with an end
+    date and drop visits between 40 weeks before end date and 12 weeks
+    after end date.
+-   `get_maternal_care_visit_dates_without_end_date()`: Uses the output
+    from `get_pregnancy_end_dates()` which identifies maternal care
+    visits *with* end dates to derive maternal care visits *without* end
+    dates. below.
+
+<!-- TODO: What is done with the mc visits without end dates then? -->
+
+<!-- TODO: Add details on how this filtering should be done -->
+
+### Glucose-lowering brand drugs for weight loss
+
+The function `exclude_purchases_weight_loss_drugs()` uses REGISTER as
+input and excludes BRANDS.
+
+<!-- TODO: Add details on how this filtering should be done -->
+
+<!-- TODO: Add data source and which brands are excluded -->
+
+### Metformin purchases for women below age 40
+
+The function `exclude_pcos()` as input to exclude all purchases of
+metformin by women below age 40 (i.e., \<= 39 years old) at the date of
+purchase.
+
+This function contains two helper functions:
+
+-   `keep_women()`
+-   `drop_age_40_below()`
+
+<!-- TODO: Add details on how this filtering should be done -->
+
+## Get initial diagnosis date
+
+Then, the first inclusion and exclusion events from the function outputs
+are dropped with `drop_first_event()` so that only those with two or
+more events are kepts. This is then used to assign an initial diagnosis
+date with `get_initial_diagnosis_date()`. Then all the outputs are
+joined together at the end.
+
+## Drop diagnosis dates outside of coverage
+
+Finally, from the raw initial diagnosis dates, those dates outside of
+the data coverage period are dropped to end with a final diagnosis date
+using `get_final_diagnosis_date()`. For details on this censoring based
```

Comment:
Hm.. There's something that's not totally clear in the naming of `get_initial_diagnosis_date()` and `get_final_diagnosis_date()`. Could this last one be a helper function `drop_diagnosis_dates_outside_coverage` inside a `get_diagnosis_date()` instead?

Reply:
I like this plan as well. Want to update these?